main.go

```diff
@@ -114,60 +112,99 @@ func main() {
 
 	eventLoopContext, eventLoopCancel := context.WithCancel(context.Background())
 
+	// Validate this service against Zebedee
 	serviceIdentityValidated := make(chan bool)
 	go func() {
 		if err = checkServiceIdentity(eventLoopContext, cfg.ZebedeeURL, cfg.ServiceAuthToken); err != nil {
 			log.ErrorC("could not obtain valid service account", err, nil)
-			eventLoopDone <- true
 		}
 		serviceIdentityValidated <- true
 	}()
 
-	eventConsumer.Start(eventLoopContext, eventLoopDone, serviceIdentityValidated)
+	// Start Event Consumer (only if it is available)
+	if serviceList.Consumer {
+		eventConsumer.Start(eventLoopContext, eventLoopDone, serviceIdentityValidated)
+	}
+
+	// Log non-fatal errors, without exiting
+	go func() {
+		var consumerErrors chan (error)
+		if serviceList.Consumer {
+			consumerErrors = syncConsumerGroup.Errors()
+		} else {
+			consumerErrors = make(chan error, 1)
+		}
+
+		var producerErrors chan (error)
```

Comment:
You could initialise both channels on the same line like so:

`var consumerErrors, producerErrors chan (error)`